### ISSUE TITLE
[Suggestion] Respawn options

### ISSUE THREAD

**Issue by Crimson-Artist** [tsunamayo/Starship-EVO#6136]
While I was testing somethings in my creative world my character suddenly died when I teleported to a NPC ship that flew into a gas giant and I hadn't had no clip mode enabled. When the loading screen was taking too long to respawn me I realized that I had completely forgotten to set my respawn point to the test base I had set up in that system and sure enough I respawned back in the starting system...on the complete opposite side of the galaxy.

Since we don't have a method of instantly teleporting to a system via admin commands (would be nice to have) and the bookmark feature is not working I couldn't get back to my base and had to go look for a new system to set up in.

In survival this could be devastating to a player for simply forgetting to set their respawn point. I propose that upon death we are presented with several options on where to respawn.
1. **Respawn at respawn point**- the normal respawn option. respawns the player at the medibay they set as their respawn point. If the respawn point is destroyed/no longer exists then the player is kicked back to the starting system. 
2. **Respawn at nearest Medibay**- Allows the player to respawn at the closest friendly space station/base that has a open medibay. This should always be in the same system as where the player died so they can resume whatever they were doing. Maybe add some kind of cost to using this option cuz it maybe too convenient than just setting a respawn point. 
3. **Fresh start**- A last resort option. Basically restart the game and respawn the player on a random planet with the same conditions as when we started the game off in with the intention being that we rebuild our way back up into space.

**Comment by tsunamayo** [comment]
Well there should be a check of whether your last medbay is valid, if not it should pick the one prior and so on. Spawn at nearest medbay could be an option indeed. 
I need to improve that part of the gameplay for sure.
Also the npc should not fly into the gaz giant, are you sure it did that? Thanks

**Comment by Crimson-Artist** [comment]
@tsunamayo on a related topic, you should really consider how you are going to handle various player destitution scenarios. The most common will be the player's ship running out of fuel and being completely dead in space. since star systems are so huge its not like the player can just float to their nearest space station and get some fuel. Same goes for planets, even the smallest moons are massive and walking to a nearby base may not be an option. I know its a part of the game for the player to keep their fuel reserves topped off but these things happen.

I had thought up a SOS mechanic that players can trigger. It would send out a ping to the entire system for help and the closest NPC ship will answer it. When the NPC arrives they will offer things like fuel and repairs at a much higher cost then at stations since you know they can. Maybe depending on the NPC's faction you could probably get it at a reduced cost or even free. The SOS mechanic can even be used maliciously to lure in victims if the player wants to be a pirate though the danger will come in the form of not being able to choose the target. A space Police ship or a much larger ship could show up. On the flipside a pirate could answer the call and end up exploiting you.

**Comment by tsunamayo** [comment]
> @tsunamayo on a related topic, you should really consider how you are going to handle various player destitution scenarios. The most common will be the player's ship running out of fuel and being completely dead in space. since star systems are so huge its not like the player can just float to their nearest space station and get some fuel. Same goes for planets, even the smallest moons are massive and walking to a nearby base may not be an option. I know its a part of the game for the player to keep their fuel reserves topped off but these things happen.
> 
> I had thought up a SOS mechanic that players can trigger. It would send out a ping to the entire system for help and the closest NPC ship will answer it. When the NPC arrives they will offer things like fuel and repairs at a much higher cost then at stations since you know they can. Maybe depending on the NPC's faction you could probably get it at a reduced cost or even free. The SOS mechanic can even be used maliciously to lure in victims if the player wants to be a pirate though the danger will come in the form of not being able to choose the target. A space Police ship or a much larger ship could show up. On the flipside a pirate could answer the call and end up exploiting you.

Yes that is something I wanted to do. Also the player should be able to answer a few of those distress call from time to time (some being traps also). I also want to give a suicide option, so you respawn at your last medbay - but I need to make sure that the players has to pay a penalty for it.

**Comment by Crimson-Artist** [comment]
> Yes that is something I wanted to do. Also the player should be able to answer a few of those distress call from time to time (some being traps also). I also want to give a suicide option, so you respawn at your last medbay - but I need to make sure that the players has to pay a penalty for it.

This is where having an aliment system would come in handy. You could just give the player a unique debuff upon respawning that either goes away with time or can only be cured with medicine
